docs: add Qiniu KODO to s3 provider list - fixes #6195

bachue opened the pull request:
<!--
Thank you very much for contributing code or documentation to rclone! Please
fill out the following questions to make it easier for us to review your
changes.

You do not need to check all the boxes below all at once, feel free to take
your time and add more commits. If you're done and ready for review, please
check the last box.
-->

#### What is the purpose of this change?

Add Qiniu KODO to s3 provider list.

#### Was the change discussed in an issue or in the forum before?

#6195 

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

ncw commented on MANUAL.html:
```diff
@@ -10592,7 +10592,7 @@ <h3 id="object-lock-enabled-s3-bucket">Object-lock enabled S3 bucket</h3>
 <p>As mentioned in the <a href="#hashes">Hashes</a> section, small files that are not uploaded as multipart, use a different tag, causing the upload to fail. A simple solution is to set the <code>--s3-upload-cutoff 0</code> and force all the files to be uploaded as multipart.</p>
 <h3 id="standard-options-3">Standard options</h3>
-<p>Here are the Standard options specific to s3 (Amazon S3 Compliant Storage Providers including AWS, Alibaba, Ceph, China Mobile, Cloudflare, ArvanCloud, Digital Ocean, Dreamhost, Huawei OBS, IBM COS, IDrive e2, Lyve Cloud, Minio, Netease, RackCorp, Scaleway, SeaweedFS, StackPath, Storj, Tencent COS and Wasabi).</p>
+<p>Here are the Standard options specific to s3 (Amazon S3 Compliant Storage Providers including AWS, Alibaba, Ceph, China Mobile, Cloudflare, ArvanCloud, Digital Ocean, Dreamhost, Huawei OBS, IBM COS, IDrive e2, Lyve Cloud, Minio, Netease, RackCorp, Scaleway, SeaweedFS, StackPath, Storj, Tencent COS, Qiniu and Wasabi).</p>
```
This file is auto generated - can you drop it from the commit please? rclone.1 and MANUAL.md

ncw commented on docs/content/s3.md:
```diff
@@ -405,17 +405,17 @@ The following characters are replaced since these are problematic when
 dealing with the REST API:
 
 | Character | Value | Replacement |
-| --------- |:-----:|:-----------:|
```
What is this change for? Can you drop it?

ncw commented on docs/content/s3.md:
```diff
@@ -2568,16 +2682,16 @@ User metadata is stored as x-amz-meta- keys. S3 metadata keys are case insensiti
-| content-type | Content-Type header | string | text/plain | N |
-| mtime | Time of last modification, read from rclone metadata | RFC 3339 | 2006-01-02T15:04:05.999999999Z07:00 | N |
-| tier | Tier of the object | string | GLACIER | **Y** |
+| Name                | Help                                                         | Type     | Example                             | Read Only |
```
Can you drop this change please?

ncw commented on docs/content/s3.md:
```diff
@@ -3338,7 +3452,7 @@ Press Enter for the default (false).
  2 / Get AWS credentials from the environment (env vars or IAM).
    \ (true)
-env_auth> 
+env_auth>
```
Please drop these changes which remove space from the end of a line - thanks

ncw commented on docs/content/s3.md:
```diff
@@ -3468,13 +3582,214 @@ So once set up, for example, to copy files into a bucket
 ### RackCorp {#RackCorp}
 
-[RackCorp Object Storage](https://www.rackcorp.com/storage/s3storage) is an S3 compatible object storage platform from your friendly cloud provider RackCorp. 
+[RackCorp Object Storage](https://www.rackcorp.com/storage/s3storage) is an S3 compatible object storage platform from your friendly cloud provider RackCorp.
```
drop...

ncw commented on backend/s3/s3.go:
```diff
@@ -2516,6 +2618,8 @@ func setQuirks(opt *Options) {
 		useMultipartEtag = false // untested
 	case "Wasabi":
 		// No quirks
+	case "Qiniu":
```
Did you try running the integration tests against Qiniu to check the quirks are correct?

From source root

- `go build ./...`
- `test_all -remotes TestQiniu:`

Wait for result to appear in your browser.

The backend test is useful to run individually - it is the most likely one to fail, 

```
cd backend/s3
go test -v -remote TestQiniu:
```

ncw reviewed (commented):
Please can you fix the check in of
- auto generated content
- random reformatting of files

See comments.

Also please can you run the integration tests to check the quirks are set correctly

Thank you

bachue commented:
> Please can you fix the check in of
> 
> * auto generated content
> * random reformatting of files
> 
> See comments.
> 
> Also please can you run the integration tests to check the quirks are set correctly
> 
> Thank you

All comments are resolved, and integration tests are passed, please review it again, thank you!

bachue commented:
@ncw sorry, is there any new comments, question or concern about this pr?

ncw commented:
Please remove the auto generated content from this commit, eg MANUAL.md

Also remove the auto generated content from s3.md

I can do this if you are having a problem.

Thanks

bachue commented:
@ncw OK, all the auto generated content have been removed from the commit. Thanks.

ncw commented:
You seem to have removed the non-auto generated parts of `docs/content/s3.md` too? Can you put those back?

bachue commented:
@ncw 

> You seem to have removed the non-auto generated parts of `docs/content/s3.md` too? Can you put those back?

@ncw Ok, they are recovered now.

ncw commented:
I've removed a bit more auto generated docs and added a section header into the s3 docs. When the CI goes green I'll merge - thank you :-)

bachue commented:
> I've removed a bit more auto generated docs and added a section header into the s3 docs. When the CI goes green I'll merge - thank you :-)

@ncw Thanks, it's green now.

ncw commented:
Thank you very much for your contribution and apologies for delays merging.
